sql/service/src/main/scala/org/apache/spark/sql/service/cli/session/SessionManager.scala

```diff
@@ -0,0 +1,346 @@
+              }
+            } else {
+              session.closeExpiredOperations()
+            }
+          })
+          sleepInterval(interval)
+        }
+      }
+
+      private def sleepInterval(interval: Long): Unit = {
+        try
+          Thread.sleep(interval)
+        catch {
+          case e: InterruptedException =>
+
+          // ignore
+        }
+      }
+    }
+    backgroundOperationPool.execute(timeoutChecker)
+  }
+
+  override def stop(): Unit = {
+    super.stop()
+    shutdown = true
+    if (backgroundOperationPool != null) {
+      backgroundOperationPool.shutdown()
+      val timeout: Long = sqlConf.getConf(ServiceConf.THRIFTSERVER_ASYNC_EXEC_SHUTDOWN_TIMEOUT)
+      try
+        backgroundOperationPool.awaitTermination(timeout, TimeUnit.SECONDS)
+      catch {
+        case e: InterruptedException =>
+          logWarning("THRIFTSERVER_ASYNC_EXEC_SHUTDOWN_TIMEOUT = " + timeout +
+            " seconds has been exceeded. RUNNING background operations will be shut down", e)
+      }
+      backgroundOperationPool = null
+    }
+    cleanupLoggingRootDir()
+  }
+
+  private def cleanupLoggingRootDir(): Unit = {
+    if (isOperationLogEnabled) {
+      try {
+        FileUtils.forceDelete(operationLogRootDir)
+      } catch {
+        case e: Exception =>
+          logWarning("Failed to cleanup root dir of HS2 logging: " +
+            operationLogRootDir.getAbsolutePath, e)
+      }
+    }
+  }
+
+  @throws[ServiceSQLException]
+  def openSession(protocol: TProtocolVersion, username: String,
+                  password: String, ipAddress: String,
+                  sessionConf: util.Map[String, String]): SessionHandle = {
+    openSession(protocol, username, password, ipAddress,
+      sessionConf, false, null)
+  }
+
+  /**
+   * Opens a new session and creates a session handle.
+   * The username passed to this method is the effective username.
+   * If withImpersonation is true (==doAs true) we wrap all the calls in ServiceSession
+   * within a UGI.doAs, where UGI corresponds to the effective user.
+   *
+   * Please see {@code ThriftCLIService.getUserName()} for
+   * more details.
+   *
+   * @param protocol
+   * @param username
+   * @param password
+   * @param ipAddress
+   * @param sessionConf
+   * @param withImpersonation
+   * @param delegationToken
+   * @return
+   * @throws ServiceSQLException
+   */
+  @throws[ServiceSQLException]
+  def openSession(protocol: TProtocolVersion, username: String, password: String,
+                  ipAddress: String, sessionConf: util.Map[String, String],
+                  withImpersonation: Boolean, delegationToken: String): SessionHandle = {
+    logWarning("create new session context")
+    logWarning(s"sqlContext = ${sqlContext}")
+    val ctx = if (sqlContext.conf.hiveThriftServerSingleSession) {
+      logWarning("single session")
+      sqlContext
+    } else {
+      logWarning("No single session")
```

Comment:
These warning logs seem excessive. Could be turn into one info level  message that contains all the info.

Reply:
> These warning logs seem excessive. Could be turn into one info level message that contains all the info.

Yea,this message is current spark thrift server show. will update .

Reply:
Removed

Reply:
Removed